src/yml_editor.py

```diff
@@ -0,0 +1,116 @@
+from typing import Dict, Optional
+import logging
+
+import six
+import yaml
+from yaml import YAMLError
+import copy
+
+
+FORMAT = '%(asctime)s -- %(levelname)s -- %(message)s '
+logging.basicConfig(format=FORMAT)
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+def nested_dict_values(d):
+    for v in d.values():
+        if isinstance(v, dict):
+            yield from nested_dict_values(v)
+        else:
+            yield v
+
+
+class YamlEditorException(Exception):
+    pass
+
+
+class YamlEditor:
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.yml_dict: Optional[Dict] = None
+
+    def __load_file(self):
+        try:
+            return open(self.file_path, 'r+')
+        except FileNotFoundError as e:
+            logger.error(f'{self.file_path} File Not Found')
+            raise e
+
+    def __load_dict(self):
+        try:
+            file = self.__load_file()
+            self.yml_dict = yaml.safe_load(file)
+        except FileNotFoundError as e:
+            raise e
+        except YAMLError as e:
+            logger.error(f'Yaml Error - {e}')
+            raise e
+
+    def add_config(self, extra_config: Dict):
+        if not self.check_extra_config(extra_config):
+            raise YamlEditorException('extra config are not using yaml datatypes')
+        if not self.yml_dict:
+            try:
+                self.__load_dict()
+            except Exception as e:
+                logger.error(f'Error while loading yaml file (file_path={self.file_path}) -- {e}')
+                raise e
+
+        try:
+            self.yml_dict = self.__merge(copy.deepcopy(self.yml_dict), extra_config)
+        except Exception as e:
+            logger.error(f'Error while merging, {e}')
+            raise e
+
+    def __merge(self, base_dict, added_dict):
+        try:
+            if base_dict is None or isinstance(base_dict, (six.string_types, float, six.integer_types)):
```

Comment:
Why did you use the same line again? do you have already function that does something similar

Reply:
There is a chance that you check if the dict is not int, float and str instead of checking that it's a dict?